vignettes/pillar.Rmd

```diff
@@ -0,0 +1,458 @@
+---
+title: "Printing vectors nicely in tibbles"
+author: "Kirill Müller, Hadley Wickham"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Printing vectors nicely in tibbles}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+options(crayon.enabled = TRUE)
+options(pillar.bold = TRUE)
+
+knitr::opts_chunk$set(collapse = TRUE, comment = pillar::style_subtle("#>"))
+
+colourise_chunk <- function(type) {
+  function(x, options) {
+    # lines <- strsplit(x, "\\n")[[1]]
+    lines <- x
+    if (type != "output") {
+      lines <- crayon::red(lines)
+    }
+    paste0(
+      '<div class="sourceCode"><pre class="sourceCode"><code class="sourceCode">',
+      paste0(
+        sgr_to_html(htmltools::htmlEscape(lines)),
+        collapse = "\n"
+      ),
+      "</code></pre></div>"
+    )
+  }
+}
+
+knitr::knit_hooks$set(
+  output = colourise_chunk("output"),
+  message = colourise_chunk("message"),
+  warning = colourise_chunk("warning"),
+  error = colourise_chunk("error")
+)
+
+# Fallback if fansi is missing
+sgr_to_html <- identity
+sgr_to_html <- fansi::sgr_to_html
+```
+
+To ensure that a vector prints nicely in a tibble, you need to understand how printing works.
+The presentation of a column in a tibble is powered by four S3 generics:
+
+* `type_sum()` determines what goes into the column header.
+* `pillar_shaft()` determines what goes into the body of the column.
+* `is_vector_s3()` and `obj_sum()` are used when rendering list columns.
+
+If you have created a vector class that can be used as a column, you can override these generics to make sure your data prints well in a tibble.
+To start, you must import the `pillar` package that powers the printing of tibbles.
+Either add `pillar` to the `Imports:` section of your `DESCRIPTION`, or call:
+
+```{r, eval = FALSE}
+usethis::use_package("pillar")
+```
+
+This short vignette assumes a package that implements a `"latlon"` vector and uses `roxygen2` to create documentation and the `NAMESPACE` file.
+To make the required methods available, we attach pillar.
```

Comment:
```suggestion
To make the required methods available in this vignette, we attach pillar. 
You don't need to do this in your package.
```

Reply:
Or maybe put this above the `use_package()` code so then you can say something like: "In a package, you instead do ..."